### DISCUSSION TITLE
Is there any way to add additional search index without algolia?

### DISCUSSION THREAD

**Discussion by shravanngoswamii** [quarto-dev/quarto-cli#9804] [Q&A] [answered]
I was making the website for Turing.jl, and they wanted to keep the documentation website and main website different so I made two repository for each website and there navbar and footer look exactly like each other, main website is served at [turinglang.org](https://turinglang.org/) through github pages and docs website is served at [turinglang.org/docs](https://turinglang.org/docs/). Now the issue is both website repos generate individual site and deploy site to there own gh-pages branch and this process generates different search.json for each site which ends up with a problem that both sites cannot search one another content!!

So what I am asking is, Is there any way to generate common search index, or maybe add one site's search index to another site in any way possible? I do not want to use algolia!

**Comment by cscheid** [comment]
> So what I am asking is, Is there any way to generate common search index, or maybe add one site's search index to another site in any way possible?

I don't think so. You could try to study the contents of the .json file and write code to merge indices, but I think you're going to run into trouble with links to another top-level site.

**Reply by shravanngoswamii** [reply]
Well, I used the jq library and its solved the issue for one site but not for another! Here is what I did:
For the main site which is being served from [`turinglang.org/`](https://turinglang.org/), I used this workflow:
It is adding docs sites original (rendered by quarto) search index to main site and added `/docs` prefix to URLs of docs search index and merged it with main sites search index and also renamed merged index as `search.json` and original as `search_original.json` and its working!
```
      - name: Render Quarto site
        run: quarto render

      - name: Rename original search index
        run: mv _site/search.json _site/search_original.json

      - name: Install jq
        run: sudo apt-get install jq

      - name: Fetch search_original.json from docs site
        run: curl -O https://raw.githubusercontent.com/TuringLang/docs/gh-pages/search_original.json

      - name: Fix URLs in docs site search index
        run: |
          jq 'map(if .href then .href = "docs/" + .href else . end) | 
              map(if .objectID then .objectID = "docs/" + .objectID else . end)' search_original.json > fixed_docs_search.json

      - name: Merge search indices
        run: |
          jq -s '.[0] + .[1]' _site/search_original.json fixed_docs_search.json > _site/search.json
```
But still this does not solve the issue for docs site as it still cannot search main sites content!
I tried to do the same process but it search URLs from [`turinglang.org/docs`](https://turinglang.org/docs) instead of directly from [`turinglang.org`](https://turinglang.org/)!
I also tried giving absolute URLs to main sites URLs in merged docs search index like this:
this is merged - https://github.com/shravanngoswamii/docs/blob/gh-pages/search.json 
  {
    "objectID": "https://turinglang.org/news/posts/2022-02-17-gsoc/index.html",
    "href": "https://turinglang.org/news/posts/2022-02-17-gsoc/index.html",
    "title": "Google Summer of Code 2022",
    "section": "",
    "text": "content........"
  },
  
but even after doing this, it searches like, `https://turinglang.org/docs/https://turinglang.org/news/posts/2022-02-17-gsoc/index.html`

Is there any other solution or any way to treat these absolute URLs as absolute one!

**Reply by shravanngoswamii** [reply]
update: This issue is also solved by adding `../` prefix using jq! Thanks for your time!

**Comment by shravanngoswamii** [comment, marked as answer]
It's solved, If any one have this common issue then you can refer these workflows:
Turing's main site: https://github.com/TuringLang/docs/blob/master/.github/workflows/publish.yml
Turing's docs site: https://github.com/TuringLang/turinglang.github.io/blob/main/.github/workflows/publish.yml
